### ISSUE TITLE
About that World class

### ISSUE THREAD

**Issue by eikofee** [Echoffee/nene-quest#4]
Following our discussions on Discord, we are thinking about adding a `World` static class between `Game` and all of the Entities. 

This class will contains all of the entities and should provide methods like checking collisions between entities and call for their handlers if so, or for the natural death of entities (out of screen for example)

It would be nice to add our ideas here before coding it.

**Comment by eikofee** [comment]
Also, instead of passing the `elapsedTime` argument on every methods, we could put a static value in `World` which updates at every frames.

**Comment by engboris** [comment]
1. Why should it be static ?
2. What do you mean by "contains" ?

Beside that, I totally agreee with these ideas.

**Comment by eikofee** [comment]
1. Because there will be only one World per game instance, we won't have any instantiation issue and it will be more organized to read imo.

2. `World` will keep the references of the entities and will handle any needed accessibility to them.

**Comment by engboris** [comment]
By "keep the references" do you mean that the entities will be stored in a container (vector, linkedlist, ...) ?

**Comment by eikofee** [comment]
Yeah, something like that.

**Comment by engboris** [comment]
Ok. I can't think of anything opposing these ideas. It seems quite safe to implement that but let's wait for a global approval.

**Comment by eikofee** [comment]
So, I started implementing the `World` class in [#6ac575](https://github.com/engboris/nene-quest/commit/6ac5752fa0ae6fe62f4c2be0b64e4e7ac5d77e05) and I didn't encountered any issue while linking entities in it. The collision system is also way easier than I thought it would it to be. 

Just a few other things to work on :
- The collisions on Z-axis, which shouldn't be much of a hassle
- Moving the player toward the mountains background seems to be glitching, but I didn't give deep thoughts on where to put the collision detection during `Player.move()`
- The bridge object also have 2 hitboxes, while other entities only have one. I altered the `Entity` class to have either one or multiple hitboxes to keep the current code compiling, but I think all of the entities should only use multiple hitboxes (`vector`, even with only one hitbox in it). Related to #2 maybe.

That said, the "main" code of `Game` can now be moved in `World`, and we should think about the `Render` class properties mentioned in #3 . Right now I think I'm able to resume the parser implementation.
